syncd

```diff
@@ -8,27 +8,93 @@
 
 CONF_FILE=syncd.conf
 
-while [ ! -e $CONF_FILE ]; do
-  if [ $PWD = "/" ]; then
-    echo "Unable to find syncd configuration file \"$CONF_FILE\"."
+usage() {
+    cat <<EOF
+	Syncd is a simple bash script that watches for file changes and rsyncs them
+    to a remote machine. It uses inotify to watch for file system changes and
+    syncs the whole directory to a remote machine using rsync.
+
+    Usage: syncd {start|stop|restart|status|run|log|help}
+
+    Commands:
+    start                                Starts the syncd daemon so file system
+                                         changes trigger rsyncing of the files
+    stop                                 Stop the syncd daemon
+    restart                              Restart the syncd daemon
+    status                               Get the status of the syncd daemon
+    run                                  Trigger a sync manually
+    log                                  Tail the daemon log
+    show-run-command                     Echo the command that will be used to
+                                         sync the files
+    help                                 Print out this help message
+    sample_config                        Print out a sample syncd.conf file
+EOF
+}
+
+sample_config() {
+cat <<EOF
+# Syncd config
+#
+# Variables:
+# - SYNCD_CONFIG_DIR: The directory in which the used syncd.conf is located.
+# - SCRIPT_DIR: The directory the syncd script has been installed to.
+
+# The directory being watched & synced.
+WATCH_DIR=$SYNCD_CONFIG_DIR/your-directory
+
+# Watch-related options
+WATCH_EXCLUDE='(\.git|___jb_)'
+# Whether to enable verbosity. If enabled, change events are output.
+WATCH_VERBOSE=0
+
+# SSH connection settings
+SSH_USER=vagrant
+SSH_HOST=hostname.local
+
+# Sync options
+REMOTE_TARGET_DIR="/var/www"
+RSYNC_EXCLUDE='sites/default/files --exclude files'
+RSYNC_OPTIONS="-Cra --out-format='[%t]--%n' --include core"
+
+# Syncd files
+LOGFILE=".syncd.log"
+PIDFILE=".syncd.pid"
+EOF
+}
+
+error_exit() {
+    usage
     exit 1
-  else
-    cd ..
-  fi
-done
+}
+
+verify_config_file_present() {
+    while [ ! -e $CONF_FILE ]; do
+        if [ $PWD = "/" ]; then
+            echo "Unable to find syncd configuration file \"$CONF_FILE\". Run with \`help\` for more information."
+            exit 1
+        else
+            cd ..
+        fi
+    done
+    . $CONF_FILE
+}
 
 SYNCD_CONFIG_DIR=$PWD
-SCRIPT=`readlink -f $0`
+if [ -L $0 ]; then
+    SCRIPT=$(readlink -f $0)
+else
+    SCRIPT=$(realpath $0)
+fi
```

Comment:
This script may not be symlinked. I copied the syncd files onto my `$PATH` so the original code broke. This uses realpath to get the full name in the event that `$0` isn't a symlink.